lib/mongo_ecto/encoder.ex

```diff
@@ -46,6 +46,8 @@ defmodule Mongo.Ecto.Encoder do
       :error       -> :error
     end
   end
+  def encode(%Mongo.Ecto.DiffEmbed{} = diff, _pk),
```

Comment:
Couldn't you make this return this:

```elixir
          Enum.map(changes, fn {key, value} ->
            {"#{field}.#{key}", value(value, pk, "update command")}
          end)
```

And then you change  `document/3` or `pair/3` inside this same module to be able to handled nested lists like these? Notice you already have a case for `{subkey, encoded}` which could be handled by the same code (i.e. if the key is a list, you merge them).

Just an idea, I am not sure if it will actually make the code better. :)